### PR TITLE
UnixPB: Script to provide GPG signature verification for downloaded binaries

### PR DESCRIPTION
**Pull request by Haroon-Khel**
- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [ ] other documentation is changed or added (if applicable)
- [ ] playbook changes run through [VPC](https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptopenjdk.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [ ] VPC/QPC not applicable for this PR
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

ref https://github.com/adoptium/infrastructure/issues/2233

Left to do

- [x] Add the rest of the public keys to variables file
- [x] Use the script in the roles which need it

Example of usage:
```
./package_signature_verification.sh -s path_to_sig_file (or -sl sigfile_url) -f path_to_binary (or -fl binary_url) -k public_key
```
I'm still undecided whether to import all of the gpg keys at the beginning of the playbook, or at each role which uses it

### CODE REVIEW

**Comment by Haroon-Khel** [comment]
Not sure if the linter errors are related to this pr

**Comment by Haroon-Khel** [comment]
The task will exit if it hits a bad signature
```
gpg: Signature made Tue Nov 19 19:36:42 2019 GMT
gpg:                using DSA key B02137D875D833D9B23392ECAE5A7FB608A0221C
gpg: Can't check signature: No public key
```
or
```
gpg: Signature made Tue Nov 19 19:36:41 2019 GMT
gpg:                using DSA key B02137D875D833D9B23392ECAE5A7FB608A0221C
gpg: BAD signature from "Robert Scholte <rfscholte@apache.org>" [unknown]
```
While it will continue on a good signature
```
gpg: Signature made Tue Nov 19 19:36:42 2019 GMT
gpg:                using DSA key B02137D875D833D9B23392ECAE5A7FB608A0221C
gpg: Good signature from "Robert Scholte <rfscholte@apache.org>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: B021 37D8 75D8 33D9 B233  92EC AE5A 7FB6 08A0 221C
```

**Review by steelhead31 — APPROVED** [review]
LGTM

**Comment by Haroon-Khel** [comment]
vpc https://ci.adoptopenjdk.net/job/VagrantPlaybookCheck/1593/console

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/group_vars/all/adoptopenjdk_variables.yml [hunk 1]
```diff
@@ -32,3 +32,12 @@ Vendor_Playbook: /Vendor_Files/Vendor_Playbook/Vendor.yml
 
 # Default BootJDK installed
 bootjdk: hotspot
+
+# GPG Public Keys
+key:
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/group_vars/all/adoptopenjdk_variables.yml:37]: Did these keys just come from the messages you get when you try to verify one without the signature being present, or were they from the upstream project's somwhere?

**Review by gdams — APPROVED** [review]

**Comment by sxa** [comment]
> vpc https://ci.adoptopenjdk.net/job/VagrantPlaybookCheck/1593/console

Can you add the latest link to the initial description of this PR please, since it currently says "not applicable" :-)

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/group_vars/all/adoptopenjdk_variables.yml (hunk 1 above)
Comment (Haroon-Khel) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/group_vars/all/adoptopenjdk_variables.yml:37]: Some had the key ids present in the same place as the binary to download. For the others I did find their fingerprint in the inital verification message. But for either case, I did search for the key on https://keyserver.ubuntu.com/ to verify its authenticity. Is this the best way to have gone about this?

**Comment by Haroon-Khel** [comment]
>

**Comment by Haroon-Khel** [comment]
vpc https://ci.adoptopenjdk.net/job/VagrantPlaybookCheck/1595/console

**Comment by Haroon-Khel** [comment]
vpc on failed platforms https://ci.adoptopenjdk.net/job/VagrantPlaybookCheck/1596/console (centos6 and solaris10)

**Comment by Haroon-Khel** [comment]
The latest vpc passes on centos 6 and solaris 10, which were the remaining problem platforms. This can be merged

**Pull request merged**